### PR TITLE
Removed deprecated code path

### PR DESCRIPTION
**Pull request by krmahadevan**

### CODE REVIEW

File: src/test/resources/yaml/a2.yaml [hunk 1]
```diff
@@ -26,8 +26,7 @@ tests:
     methodSelectors:
       - className: org.testng.internal.XmlMethodSelector
         priority: 1
-      - expression: groups.containsKey("test1")
-        language: beanshell
+      - script: { expression: groups.containsKey("test1"), language: beanshell }
```
Comment (juherr) [inline src/test/resources/yaml/a2.yaml:29]: The yaml format is not supposed to change but some deprecated methods were used by the yaml library.
Could you check how it is possible to customize the yaml serialization?

**Review by juherr — CHANGES_REQUESTED** [review]

File: src/main/java/org/testng/TestNG.java [hunk 2]
```diff
@@ -419,15 +418,11 @@ public void setThreadCount(int threadCount) {
   @Deprecated
   // TODO: krmahadevan: This method is being used by Gradle. Removal causes build failures.
```
[the remark this answers is not in the document]
Comment (krmahadevan) [reply src/main/java/org/testng/TestNG.java:419]: @juherr - Yes I did. Here's the issue i raised a long time back https://github.com/gradle/gradle/issues/10507

File: src/test/resources/yaml/a2.yaml (hunk 1 above)
Comment (krmahadevan) [reply src/test/resources/yaml/a2.yaml]: Fixed and pushed.

**Comment by krmahadevan** [comment]
@juherr - I have fixed the review comments. please take a look

File: src/main/java/org/testng/ITestNGMethod.java [hunk 3]
```diff
@@ -172,34 +170,15 @@ default boolean hasAfterGroupsConfiguration() {
...
-   * Please use {@link #setRetryAnalyzerClass(Class)} instead.
-   *
-   * @param retryAnalyzer The retry analyzer
-   */
-  @Deprecated
-  void setRetryAnalyzer(IRetryAnalyzer retryAnalyzer);
-
   default IRetryAnalyzer getRetryAnalyzer(ITestResult result) {
-    return getRetryAnalyzer();
+    return null;
```
Comment (juherr) [inline src/main/java/org/testng/ITestNGMethod.java:174]: If you remove the deprecated method, you assume users will have to use the new one. In that case, there is no more default implementation.

File: src/main/java/org/testng/annotations/ITestAnnotation.java [hunk 4]
```diff
@@ -74,19 +75,10 @@
...
-   *
-   * @return The retry analyzer
-   */
-  @Deprecated
-  IRetryAnalyzer getRetryAnalyzer();
-
   void setRetryAnalyzer(Class<? extends IRetryAnalyzer> c);
 
   default Class<? extends IRetryAnalyzer> getRetryAnalyzerClass() {
-    return getRetryAnalyzer().getClass();
```
Comment (juherr) [inline src/main/java/org/testng/annotations/ITestAnnotation.java:89]: Just remove the default implementation

**Review by juherr — COMMENTED** [review]
Just a last remark on the deprecation strategy: empty default methods hide the need for a method to implement (which is an objective of interfaces). Otherwise, everything will run but the behavior won't be the expected one.

File: src/main/java/org/testng/ITestNGMethod.java (hunk 3 above)
Comment (krmahadevan) [reply src/main/java/org/testng/ITestNGMethod.java]: Fixed

File: src/main/java/org/testng/annotations/ITestAnnotation.java (hunk 4 above)
Comment (krmahadevan) [reply src/main/java/org/testng/annotations/ITestAnnotation.java:89]: Fixed

**Comment by krmahadevan** [comment]
@juherr - Have fixed all the comments. Please take another look

**Review by juherr — APPROVED** [review]

**Comment by missedone** [comment]
@krmahadevan , is the SNAPSHOT published anywhere? i'd like to test it out on testng-remote and testng-eclipse projects. thx

**Comment by krmahadevan** [comment]
@missedone - I am not sure if we are publishing snapshots anymore to anywhere.
@cbeust @juherr can you please advise on this? I did see a github action but that seems to be only for releases. Not sure if it can be used for snapshots as well. It would be great if we can have a github action that uploads snapshots also to Maven central.

**Comment by juherr** [comment]
It looks broken: the latest snapshot is `7.1.1 - Fri Aug 07 2020` https://oss.sonatype.org/content/repositories/snapshots/org/testng/testng/

**Pull request merged**